pkg/controller/subscription/webapp/main.go

```diff
@@ -13,12 +13,17 @@ import (
 	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type UpgradeNotifier struct {
+type UpgradeNotifier interface {
```

Comment:
Why is this stuff in here?

Reply:
I can answer this one. As @jjaferson added unit tests that would hit the `NotifyUpgrade` call, this would fail the unit tests as the subscription controller was initially instantiating the `UpgradeNotifier` struct with a `rest.Config` object, which can't be mocked in unit tests. So we had to refactor it here to make `UpgradeNotifier` an interface that can be mocked in the unit tests